nova/tests/unit/objects/test_instance.py

```diff
@@ -1869,13 +1869,24 @@ def test_with_fault(self, mock_get_all, mock_fault_get):
         mock_fault_get.assert_called_once_with(self.context,
             [x['uuid'] for x in fake_insts])
 
+    @mock.patch('nova.objects.InstanceMappingList.get_by_instance_uuids')
     @mock.patch.object(db, 'instance_fault_get_by_instance_uuids')
-    def test_fill_faults(self, mock_fault_get):
+    def test_fill_faults(self,
+                         mock_fault_get,
+                         mock_get_inst_map_list):
         inst1 = objects.Instance(uuid=uuids.db_fault_1)
         inst2 = objects.Instance(uuid=uuids.db_fault_2)
         insts = [inst1, inst2]
         for inst in insts:
             inst.obj_reset_changes()
+
+        mock_get_inst_map_list.return_value = [
+            mock.Mock(cell_mapping=None,
+                      instance_uuid=uuids.db_fault_1),
+            mock.Mock(cell_mapping=None,
```

Comment:
Both your instances have the same cell here. If we want to upstream this - and we should - we need a proper test-case. Best thing is to have a test-case that can be run without your changes, but fails, so we can convince upstream that there's a bug. It should then work with your changes obviously ;)

Reply:
Done. Improved test to cover different cells.